lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/TypeWithKind.java

```diff
@@ -39,13 +39,14 @@ boolean isValidGuestType() {
      */
     private final static List<String> validGuestTypes =
             List.of(
+                    "org.enso.interpreter.runtime.data.EnsoObject",
```

Comment:
Wow, this is wild. List of allowed types. For now I can just extend it with `EnsoObject` interface, but we should use `processEnv.getTypes()` to check the real types and not just names.

Reply:
Good idea. We can certainly do that. Created https://github.com/enso-org/enso/issues/7552 to track that.